filters/accesslog/control.go

```diff
@@ -13,46 +13,70 @@ const (
 	AccessLogEnabledKey = "statebag:access_log:proxy:enabled"
 )
 
+// Common filter struct for holding access log state
+type AccessLogFilter struct {
+	Enable   bool
+	Prefixes []int
+}
+
+func (al *AccessLogFilter) Request(ctx filters.FilterContext) {
+	bag := ctx.StateBag()
+	bag[AccessLogEnabledKey] = al
+}
+
+func (*AccessLogFilter) Response(filters.FilterContext) {}
+
+func extractFilterValues(args []interface{}, enable bool) (filters.Filter, error) {
+	prefixes := make([]int, 0)
+	for _, prefix := range args {
+		intPref, ok := 0, false
+		switch prefix.(type) {
```

Comment:
We should not rely on the eskip parser returning only float64, IMO better to test while creating the filter for all these types.

Reply:
This is not how we created many of the predicates and filters until now. If this is not documented as such, we assumed `float64` as the specification. That was the base for not advancing with https://github.com/zalando/skipper/pull/430 some time ago.

Reply:
i think it's fine to allow and check for integers, because it helps with programmatically defined eskip objects, e.g. during testing. On the other hand, I would return an error when an argument is not a number.

Reply:
Ah. Didn't know this. I just assumed it was returning float64 because I'm running a 64 bit system. I can leave it in if it helps with testcases.